### ISSUE TITLE
Rearrange login screen

### ISSUE THREAD

**Issue by mwinokan** [m2ms/fragalysis-frontend#1566]
@phraenquex suggests rearranging the login screen to make the keycloak login less prominent / confusing for users

![Screenshot 2024-11-07 at 11 11 13](https://github.com/user-attachments/assets/f9325386-2c9c-42ce-baf3-727791bb2008)

- [ ] ORCID should be removed
- [ ] Keycloak login should be hidden

@tdudgeon please investigate if this is doable on the backend, otherwise we may need a frontend solution

**Comment by mwinokan** [comment]
Dependent on #1451

**Comment by phraenquex** [comment]
@tdudgeon after the upgrade, this login screen is now even worse:  CAS is now almost invisible.  So moving it higher up the queue.  

(cc @mwinokan )

![Image](https://github.com/user-attachments/assets/f3aa6230-6a6b-4753-ae75-eb351f6d9325)

**Comment by mwinokan** [comment]
@tdudgeon says the mechanics of how to customise this screen will need to be determined in this ticket.

Then a spin-out ticket for the f/e will be needed for @matej-vavrek to work on

**Comment by alanbchristie** [comment]
Keycloak login customisation has now been explored in our [docker-keycloak](https://github.com/InformaticsMatters/docker-keycloak) repository where it has now been enhanced to illustrate how to customise it.

The Keycloak we use is built (customised) in this repository, and it has now been updated with an example and a docker-compose file to allow local deployment and customisation. There are brief notes in the `README` and the `docker-compose.yaml` file but customisation documentation is the responsibility of keycloak's own documentation.

I suggest we **fork** this repository and create a **customer** branch for each customer's customisation.

**Comment by mwinokan** [comment]
@tdudgeon please point @matej-vavrek to where he can customise the HTML and CSS and then @matej-vavrek please see how easy it would be to change the order and visual priority of the different login options.

Ideally the login dialog for keycloak is hidden behind a toggle at first as it is an admin feature and the CAS option is more prominent. 

@phraenquex says that ORCID can also be hidden as at this point to log in to Fragalysis you need a FedID (or keycloak account)

**Comment by tdudgeon** [comment]
ORCID option has been removed from the login screen. It can easily be re-enabled.
There are 71 users who have logged in using ORCID.

**Comment by matej-vavrek** [comment]
Current form:

![Image](https://github.com/user-attachments/assets/45bd90c3-3838-4ecb-813f-98753f502b58)

Tbd:

![Image](https://github.com/user-attachments/assets/fb03a87b-a646-4222-8142-62c2ffe92eab)

Basically, we can customize it as we wish.
I suggest to move bottom part (1) with other login providers above login form (2). Remove "Or sign in with" from (1) or move it out to be, for example, a toggle to show the login form (2), i.e. there would be list of other login providers ("Diamond CAS", ..), bellow will be this text like "Or sign in with Keycloak login" which would be toggle to show/hide the login form (2) [hidden by default]. Also we can have some highlight flag for an other login provider to show it in its own separate line and more visually appealing.

This way we can rearrange it simply with changing layout a bit and adding some javascript in custom theme without touching any actual Keycloak code.

I can prepare some showcase for better visualisation.

**Comment by mwinokan** [comment]
Thanks @matej-vavrek

I definitely agree with your points. A nice big Diamond CAS button at the top, and the keycloak login hidden behind a toggle will be great

**Comment by matej-vavrek** [comment]
@mwinokan I have just reused button styling

![Image](https://github.com/user-attachments/assets/050b26e4-85ed-471f-951d-0cfcb6bb4877)

![Image](https://github.com/user-attachments/assets/552c76d6-ad06-42df-875d-481727d77a15)

**Comment by mwinokan** [comment]
That's great @matej-vavrek 

Please could you change the text to: "Or sign in with Keycloak"

**Comment by matej-vavrek** [comment]
@mwinokan, yes, we can use our own text. Hmm should be there also the info about "the click" like "Or press here to sign in with your Keycloak account" or something like that? Or should the login form automatically roll out on hover?

**Comment by mwinokan** [comment]
@matej-vavrek sure, "click here to sign in with keycloak" will be good, thanks

**Comment by matej-vavrek** [comment]
@alanbchristie, where to push changes? Should we have own fork or [IM repo](https://github.com/InformaticsMatters/docker-keycloak) is also valid option?

**Comment by alanbchristie** [comment]
Two possibilities - 1) build your own image (simple) or 2) inject custom files into a "base image" (needs playbook changes).

## 1
Fork the IM repo to something like `xchem/fragalysis-keycloak`. This way each "customer"/"client" can create and manage their own authentication solution using whatever built-in configuration they desire. They don't need to modify the "reference" repo, and are in full control.

Then you just deploy `xchem/keycloak:26.0.5` rather than `informaticsmatters/keycloak:26.0.5`.

## 2
The other route is to simply apply the configuration (JavaScript or config files) to the existing base image, using what Kubernetes calls a **ConfigMap**. These are used to insert files into a container. But if the config becomes complex, spreading over many files/directories this will become complicated and not just complex. So, if the modification is just one file then this might be a better approach - and we keep the base container in one repo.

This second approach will require modification the the playbooks (to add the **ConfigMap**) but I'll need to experiment with it.

Cautionary note ... the is that, as time goes by, it's never "just one file"!

**Comment by matej-vavrek** [comment]
I have updated/created custom theme which is seven files in total at the moment.

![Image](https://github.com/user-attachments/assets/2f8b28b1-789c-40e8-9f33-72f96a55d46a)

**Comment by phraenquex** [comment]
@matej-vavrek is that option 1 (from what @alanbchristie suggested)?  Certainly #1 strikes me as the better idea, since we do need this future-proofed beyond the Diamond installation.

**Comment by matej-vavrek** [comment]
@phraenquex, @alanbchristie, I think it is the option 1

**Comment by mwinokan** [comment]
@alanbchristie & @matej-vavrek please deploy the custom theme using option 1

**Comment by matej-vavrek** [comment]
@tdudgeon, @alanbchristie to [to highlight an identitiy provider link](https://github.com/xchem/fragalysis-keycloak?tab=readme-ov-file#custom-theme-settings)  and show it as a button suffix its alias with `-highlight` or name the alias as `diamond` in "General settings" of a provider

**Comment by alanbchristie** [comment]
Although `xchem/keycloak:26.0.5.1` has been deployed I am (atm) unable to change the theme (the SAVE button appears not to respond). Am investigating...

**Comment by mwinokan** [comment]
Thanks @alanbchristie and @matej-vavrek. Please deploy this to staging (and production if that needs to be done together).
